src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int GROWTH_FACTOR = 2;
+    private T[] stack;
+    private int size;
+    private int capacity;
 
+    public MateStack() {
+        final int default_capacity = 10;
+        stack = (T[]) new Object[default_capacity];
+        size = 0;
+        capacity = default_capacity;
```

Comment:
Since `default_capacity` is only used once and is a constant, you could remove the local variable and initialize `capacity` directly with the value, or define it as a static final field.